src/protocolgame.cpp

```diff
@@ -1977,6 +2043,161 @@ void ProtocolGame::sendMarketDetail(uint16_t itemId)
 	writeToOutputBuffer(msg);
 }
 
+void ProtocolGame::sendStoreError(StoreError_t errorType, const std::string& message) {
+	NetworkMessage msg;
+	msg.addByte(0xE0);
+	msg.addByte(errorType);
+	msg.addString(message);
+
+	writeToOutputBuffer(msg);
+}
+
+void ProtocolGame::requestPurchaseData(uint32_t offerId, StoreOfferType_t offerType) {
+	NetworkMessage msg;
+	msg.addByte(0xE1);
+	msg.add<uint32_t>(offerId);
+	msg.addByte(offerType);
+
+	writeToOutputBuffer(msg);
+}
+
+void ProtocolGame::sendCoinBalance() {
+	NetworkMessage msg;
+	msg.addByte(0xF2);
+	msg.addByte(0x01);
+	msg.addByte(0xDF);
+	msg.addByte(0x01);
+
+	msg.add<uint32_t>(player->coinBalance); //total coins
+	msg.add<uint32_t>(player->coinBalance); //transferable coins
+
+	writeToOutputBuffer(msg);
+}
+
+void ProtocolGame::updateCoinBalance() {
+	NetworkMessage msg;
+	msg.addByte(0xF2);
+	msg.addByte(0x00);
+
+	writeToOutputBuffer(msg);
+
+	g_dispatcher.addTask(
+		createTask(std::bind([](ProtocolGame_ptr client) {
```

Comment:
Hey guys, I'm not very familiar with The forgotten server structure, but has been given a "task" to fix this system you work on here. I just went through pieces of code and I wonder if it's such a good idea to pass ProtocolGame_ptr to the dispatcher??